lib/auth/auth.js

```diff
@@ -38,20 +38,33 @@ auth.check = (request, log, awsService, data) => {
         // handle temporary security credentials
         if (authHeader.startsWith('AWS ')) {
             log.trace('authenticating request with auth v2 using headers');
-            return authV2.headerAuthCheck.check(request, log, data);
+            const res = authV2.headerAuthCheck.check(request, log, data);
+            res.authType = 'REST-HEADER';
```

Comment:
Yeah, I agree. @LaurenSpiegel If you want to keep the `headerAuthCheck.check` pure from this logic, you can make a utility function in the same file which deals with this logic.